### PR TITLE
feat: values aggregates for events

### PR DESCRIPTION
**Pull request by BugGambit**

### CODE REVIEW

File: src/types/types.ts [hunk 1]
```diff
@@ -713,15 +718,28 @@ export type DeleteAssetMapping3D = AssetMapping3DBase;
...
 
+export interface EventAdvancedAggregate extends EventCountAggregate {
+  /**
+   * The field name(s) to apply the aggregation on. Currently limited to one field.
+   */
+  fields: ('type' | 'subtype' | 'dataSetId')[];
+  /**
+   * Type of aggregation to apply. values - Get unique values (upto max 1000) in the specified field ordered by frequency.
+   */
+  aggregate: 'values';
```
Comment (hhenriques) [inline src/types/types.ts]: Did you consider `uniqueValues` instead of `values`? Think it's more explicit

File: src/types/types.ts [hunk 2]
```diff
@@ -713,15 +718,28 @@ export type DeleteAssetMapping3D = AssetMapping3DBase;
 export type EXECUTE = 'EXECUTE';
 
 /**
- * Query schema for asset aggregate endpoint
+ * Query schema for event aggregate endpoint
  */
-export interface EventAggregateQuery {
+export type EventAggregateQuery = EventCountAggregate | EventAdvancedAggregate;
+
+export interface EventCountAggregate {
```
Comment (hhenriques) [inline src/types/types.ts]: Why the change in name here? The `filter` is generic to all aggregates, not only `count` right?

File: src/types/types.ts (hunk 1 above)
Comment (BugGambit) [reply src/types/types.ts]: The schema is defined here: https://docs.google.com/document/d/1xrts3tu_7ldJDNiY6RJQ4Qt48onSC5R-8pmtiM9MViA/edit#bookmark=id.5x5u3i4p67l4

Comment (hhenriques) [reply src/types/types.ts]: `values` seems a bit generic though. I first reviewed the test and just looking at the code, I couldn't understand what the result of the request was

File: src/types/types.ts (hunk 2 above)
Comment (BugGambit) [reply src/types/types.ts]: Fixed

File: src/types/types.ts [hunk 3]
```diff
@@ -713,15 +718,30 @@ export type DeleteAssetMapping3D = AssetMapping3DBase;
...
   filter?: EventFilter;
 }
 
+export interface EventCountAggregate extends EventBaseAggregate {}
+
+export interface EventAdvancedAggregate extends EventBaseAggregate {
+  /**
+   * The field name(s) to apply the aggregation on. Currently limited to one field.
+   */
+  fields: ('type' | 'subtype' | 'dataSetId')[];
```
Comment (polomani) [inline src/types/types.ts:736]: can we add a code snippet in the `eventsApi.ts:aggregate(...)` ?
even I don't understand how it works from the first glance

Comment (hhenriques) [reply src/types/types.ts:736]: @polomani the plan is to have a function per aggregate type (so a single API can spawn multiple SDK functions). Not sure how snippets would work there

**Review by BugGambit — COMMENTED** [review]
Awaiting @ponnapz api change for this one to pass the CI and be merged

File: src/types/types.ts (hunk 3 above)
Comment (polomani) [reply src/types/types.ts:736]: whose plain is that?
so far we had one function per endpoint why do you want to change that?
this will introduce a huge gap between api schemas and typescript types
what are the reasons?

File: src/resources/events/eventsApi.ts [hunk 4]
```diff
@@ -56,11 +58,28 @@ export class EventsAPI extends BaseResourceAPI<CogniteEvent> {
...
+  /**
+   * [Aggregate events](https://docs.cognite.com/api/v1/#operation/aggregateEvents)
+   *
+   * ```js
+   * const aggregates = await client.events.uniqueValuesAggregate({ filter: { assetIds: [1, 2, 3] }, fields: ['subtype'] });
+   * console.log('Unique values: ', aggregates)
+   * ```
+   */
+  public uniqueValuesAggregate = (
+    query: EventUniqueValuesAggregate
```
Comment (polomani) [inline src/resources/events/eventsApi.ts:75]: this is a bit messy, I don't think we can have it.
it's not java, we don't need to complicate things

Comment (polomani) [reply src/resources/events/eventsApi.ts:75]: also this kills any idea of having even semi generated sdks

Comment (BugGambit) [reply src/resources/events/eventsApi.ts:75]: Why? Without it then the return type is really messy since it is dependent on the function call.
We discussed also the semi generated sdks scenario with this approach with @hhenriques

**Comment by polomani** [comment]
https://github.com/cognitedata/cognite-sdk-js/pull/378
maybe it doesn't defend my argument about generated sdks, but still needs your opinion

**Comment by polomani** [comment]
I closed that alternative pr.
I think we should change it to `client.events.aggregate.uniqueValues(...)` when we do a breaking change

File: src/types/types.ts [hunk 5]
```diff
@@ -191,13 +191,22 @@ export interface Asset
   parentExternalId?: CogniteExternalId;
 }
 
-export interface AggregateResponse {
+export type AggregateResponse = CountAggregateResponse;
```
Comment (polomani) [inline src/types/types.ts]: I don't think we should introduce aliases.
can't we just use AggregateResponse directly?

File: src/types/types.ts [hunk 6]
```diff
@@ -713,15 +722,26 @@ export type DeleteAssetMapping3D = AssetMapping3DBase;
...
+export interface EventBaseAggregate {
   /**
    * Filter on events with strict matching.
    */
   filter?: EventFilter;
 }
 
+export interface EventCountAggregate extends EventBaseAggregate {}
+
+export interface EventUniqueValuesAggregate extends EventBaseAggregate {
```
Comment (polomani) [inline src/types/types.ts]: same here, too many types

**Review by polomani — COMMENTED** [review]
I would suggest to remove aliases, other than that looks good

File: src/types/types.ts (hunk 3 above)
Comment (polomani) [reply src/types/types.ts:736]: ok

**Review by polomani — APPROVED** [review]

**Comment by polomani** [comment]
when squash
`feat(events): unique values aggregate`

**Review by polomani — APPROVED** [review]
